docker/lib/dependabot/docker/version.rb

```diff
@@ -9,13 +9,21 @@ module Docker
     # See https://www.oracle.com/java/technologies/javase/versioning-naming.html
     # for a description of Java versions.
     #
-    class Version < Gem::Version
+    class Version < Dependabot::Version
       def initialize(version)
         release_part, update_part = version.split("_", 2)
 
-        @release_part = Gem::Version.new(release_part.tr("-", "."))
+        @release_part = Dependabot::Version.new(release_part.tr("-", "."))
 
-        @update_part = Gem::Version.new(update_part&.start_with?(/[0-9]/) ? update_part : 0)
+        @update_part = Dependabot::Version.new(update_part&.start_with?(/[0-9]/) ? update_part : 0)
+      end
+
+      def self.correct?(version)
+        super(new(version).to_semver)
+      end
+
+      def to_semver
+        @release_part.to_semver
```

Comment:
how would this handle, for example, `11.0.16.1-jdk` where there is no `_` character to separate the `release_part` and `update_part`? 

Seems like it would get turned into `11.0.16.1.jdk`, but does that work correctly when comparing versions?

Reply:
I think that relates to this issue: https://github.com/dependabot/dependabot-core/issues/5911, and to the comments that led me into opening it. See https://github.com/dependabot/dependabot-core/pull/5886/files#diff-fc43c2354fbec2f5957f7cd35c7b6e7f87e3372fe8cf2b9d50bddc512545889d.

Basically, `Docker::Version` currently only deals with numeric versions. Dealing with prerelease segments and stuff like that is handled by the `UpdateChecker`.

Reply:
Thanks for the links and TIL that we split this logic 🙈  

The regex we use to capture the suffix is in https://github.com/dependabot/dependabot-core/blob/5ae1a8c240a1cdc45545672200ac8f0f80323094/docker/lib/dependabot/docker/update_checker.rb#L45-L58